packages/core/src/fhirpath/utils.ts

```diff
@@ -101,20 +101,29 @@ function getTypedPropertyValueWithSchema(
   let resultValue: any = undefined;
   let resultType = 'undefined';
 
-  if (types.length === 1) {
-    resultValue = input.value[path];
+  if (path === 'value[x]') {
```

Comment:
This probably should be `path.endsWith('[x]')` or just `types.length > 0`

(`types.length` should always be greater than zero if `path.endsWith('[x]')`)